### PR TITLE
nixos/suricata: Fix module and add to module-list

### PR DESCRIPTION
**Pull request by felbinger**
I forgot to add the suricata module (https://github.com/NixOS/nixpkgs/pull/313671) to the modules-list, so it doesn't appear in the search.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by bjornfor** [comment]
You might want to remove `import = [ [...] ../modules/services/networking/suricata/default.nix ]` from nixos/tests/suricata.nix (in a separate commit).

File: nixos/modules/services/networking/suricata/default.nix [hunk 1]
```diff
@@ -22,7 +22,10 @@ let
     ;
 in
 {
-  meta.maintainers = with lib.maintainers; [ felbinger ];
+  meta = {
+    maintainers = with lib.maintainers; [ felbinger ];
+    buildDocsInSandbox = false;
```
Comment (bjornfor) [inline nixos/modules/services/networking/suricata/default.nix:27]: Only 43 hits for `git grep "buildDocsInSandbox = false"` in all of nixpkgs. I don't think that option should be set lightly.

I suspect things like `default = "${cfg.package}/etc/suricata/reference.config";` in `nixos/modules/services/networking/suricata/settings.nix` is why it fails. Try using `defaultText = ...` and make it build with the default settings?

Comment (felbinger) [reply nixos/modules/services/networking/suricata/default.nix]: I agree, seams like these two default values we're responsible. I adjusted it as you suggested, thanks!

File: nixos/modules/services/networking/suricata/settings.nix [hunk 2]
```diff
@@ -603,12 +702,14 @@ in
     "reference-config-file" = mkOption {
       type = types.str;
       default = "${cfg.package}/etc/suricata/reference.config";
+      defaultText = "/etc/suricata/reference.config";
```
Comment (bjornfor) [inline nixos/modules/services/networking/suricata/settings.nix:705]: defaultText should represent the default, not something else. Please try `defaultText = "\${config.services.suricata.package}/etc/suricata/reference.config"`.

**Comment by bjornfor** [comment]
Commit "[nixos/suricata: add module to modules-list](https://github.com/NixOS/nixpkgs/pull/349826/commits/83c2bc9c00f32b1442a78bc58b052dee7dcb091a)" is now doing a lot more. I suggest to break this PR down into these commits:

1. Fix module
2. Add module to module-list.nix
3. Remove now unneeded module in nixosTests

**Comment by bjornfor** [comment]
Please mind the order of commits. Each commit should be atomic and take the code from one working state to the next. Currently, the first commit would break the test.

**Comment by hax404** [comment]
You may reference the test in https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/ids/suricata/default.nix

**Review by SuperSandro2000 — COMMENTED** [review]
the description could be a bit more elaborate and more a full sentence but other than that looking good

File: nixos/tests/suricata.nix [hunk 3]
```diff
@@ -8,7 +8,6 @@ import ./make-test-python.nix (
       ids = {
         imports = [
           ../modules/profiles/minimal.nix
-          ../modules/services/networking/suricata/default.nix
```
Comment (Ma27) [inline nixos/tests/suricata.nix:11]: Any reason why `minimal` is still here?

**Review by Ma27 — DISMISSED** [review]
> You might want to remove import = [ [...] ../modules/services/networking/suricata/default.nix ] from nixos/tests/suricata.nix (in a separate commit).

WHy in a separate commit? This gives you one commit where the test doesn't evaluate.

> the description could be a bit more elaborate and more a full sentence but other than that looking good

I don't think single-word descriptions are appropriate, agreed.

---

Also, I noticed that the module has some import-from-derivation which made me build erlang while evaluating `nixos/release.nix`. Please remove that as well.

---

Also I must wonder why this got merged in this state in the first place :thinking:

**Comment by bjornfor** [comment]
>> You might want to remove import = [ [...] ../modules/services/networking/suricata/default.nix ] from nixos/tests/suricata.nix (in a separate commit).
>
> WHy in a separate commit? This gives you one commit where the test doesn't evaluate.

I suggested a commit order here: https://github.com/NixOS/nixpkgs/pull/349826#issuecomment-2424133221 (no breakage).

File: nixos/tests/suricata.nix (hunk 3 above)
Comment (felbinger) [reply nixos/tests/suricata.nix:11]: you right, not required; removed it.

**Comment by felbinger** [comment]
> > the description could be a bit more elaborate and more a full sentence but other than that looking good
>
> I don't think single-word descriptions are appropriate, agreed.

Not quiet sure what to put in there yet. Some of them have good descriptions in https://docs.suricata.io/en/suricata-7.0.7/configuration/suricata-yaml.html, which I am going to adopt, but some of them aren't really documented that well.

**Comment by bjornfor** [comment]
Commit `nixos/suricata: add module to modules-list` contains patch hunks that belong to the other (first) commit.

File: nixos/modules/services/networking/suricata/settings.nix [hunk 4]
```diff
@@ -97,6 +172,9 @@ in
         }
       );
       default = { }; # add default values to config
+      description = ''
+        Variables to be used within the suricata rules
```
Comment (bjornfor) [inline nixos/modules/services/networking/suricata/settings.nix:176]: Missing dot at end of sentence.

Technically, all the above variable options should also end with a dot.

File: nixos/modules/services/networking/suricata/settings.nix [hunk 5]
```diff
@@ -540,10 +647,16 @@ in
             enabled = mkOption {
               type = types.either types.bool (types.enum [ "auto" ]);
               default = "auto";
+              description = ''
+                Enable unix-command socket
```
Comment (bjornfor) [inline nixos/modules/services/networking/suricata/settings.nix:651]: Missing dot.

**Review by bjornfor — APPROVED** [review]

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**